.pullapprove.yml

```diff
@@ -0,0 +1,146 @@
+#  users_unavailable: []
+
+# Meta field that goes unused by PullApprove to allow for defining aliases to be
+# used throughout the config.
+meta:
+  # Note: Because all inactive groups start as pending, we are only checking pending and rejected active groups.
+  no-groups-above-this-pending: &no-groups-above-this-pending len(groups.active.pending.exclude("required-minimum-review")) == 0
+  no-groups-above-this-rejected: &no-groups-above-this-rejected len(groups.active.rejected.exclude("required-minimum-review")) == 0
+
+  defaults: &defaults
+    reviews:
+      # Authors provide their approval implicitly, this approval allows for a reviewer
+      # from a group not to need a review specifically for an area of the repository
+      # they own.  This is coupled with the `required-minimum-review` group which requires
+      # that all PRs are reviewed by at least one team member who is not the author of
+      # the PR.
+      author_value: 1
+
+# turn on 'draft' support
+# https://docs.pullapprove.com/config/github-api-version/
+# https://developer.github.com/v3/previews/#draft-pull-requests
+github_api_version: 'shadow-cat-preview'
+
+# https://docs.pullapprove.com/config/overrides/
+# Note that overrides are processed in order.
+overrides:
+  # For PRs which are still being worked on, either still in draft mode or indicated through WIP in
+  # title or label, PullApprove stays in a pending state until its ready for review.
+  - if: "draft or 'WIP' in title or 'PR state: WIP' in labels"
+    status: pending
+    explanation: 'Waiting to send reviews as PR is WIP'
+  # Disable PullApprove on specific PRs by adding the `PullApprove: disable` label
+  - if: "'PullApprove: disable' in labels"
+    status: success
+    explanation: "PullApprove skipped because of 'PullApprove: disable' label"
+  # If no file matching based groups are active, report this pull request as failing. Most likely,
+  # the PR author would need to update the PullApprove config, or create new group.
+  - if: len(groups.active) == 0
+    status: failure
+    explanation: 'At least one group must match this PR. Please update an existing review group, or create a new group.'
+
+groups:
+  components-team:
+    <<: *defaults
+    conditions:
+      - >
+        contains_any_globs(files, [
+          '.editorconfig',
+          '.prettierignore',
+          '.prettierrc',
+          '.stylelintrc.json',
+          'BUILD.bazel',
+          'CHANGELOG_ARCHIVE.md',
+          'CHANGELOG.md',
+          'CODE_OF_CONDUCT.md',
+          'CODE_REVIEWS.md',
+          'CODING_STANDARDS.md',
+          'CONTRIBUTING.md',
+          'DEV_ENVIRONMENT.md',
+          'FAQ.md',
+          'MODULE.bazel.lock',
+          'MODULE.bazel',
+          'package.json',
+          'packages.bzl,',
+          'packages.bzl',
+          'pkg-externals.bzl',
+          'pnpm-lock.yaml',
+          'pnpm-workspace.yaml',
+          'README.md',
+          'SECURITY.md',
+          'tslint.json',
```

Comment:
I'm not a fan of listing out all these top-level files, but I'm not sure if there's a way to:
1. Have the `components-team` own all files by default.
2. Have some infra-specific files be owned by the `components-infra` team.

Reply:
You could have a rule for `components-team` that doesn't have any file matching conditions, but has a condition that no other rule above it matches.

The rules are processed in order in the yaml, so if you have a `components-infra` group that matches it, you can then later just have the other rule match "everything that isn't already matched"

I think...